homeassistant/components/azure_devops/__init__.py

```diff
@@ -56,17 +60,36 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         entry.data[CONF_PROJECT],
     )
 
-    async def async_update_data() -> list[DevOpsBuild]:
+    async def async_update_data() -> tuple[
+        list[DevOpsBuild], list[DevOpsWorkItemValue] | None
+    ]:
         """Fetch data from Azure DevOps."""
-
-        try:
-            return await client.get_builds(
-                entry.data[CONF_ORG],
-                entry.data[CONF_PROJECT],
-                BUILDS_QUERY,
-            )
-        except (aiohttp.ClientError, aiohttp.ClientError) as exception:
-            raise UpdateFailed from exception
+        async with async_timeout.timeout(20):
+            try:
+                builds: list[DevOpsBuild] = await client.get_builds(
+                    entry.data[CONF_ORG],
+                    entry.data[CONF_PROJECT],
+                    BUILDS_QUERY,
+                )
+                if pat is not None:
+                    wiql_result: DevOpsWiqlResult = await client.get_work_items_ids_all(
```

Comment:
Even if you have a PAT configured, that does not mean that the PAT (your user) have access to work items for the org/project that is configured.

I do not see how that is handled here, is that done in the lib and it just returns `None` when you don't have access to the resource?

Reply:
Should this even be in the same coordinator?
It provides separate data from builds, and does not use build information.

The great thing about splitting coordinators is that if all entities that use a coordinator is disabled, the coordinator will not call the API to update the data as there is no need for it.

Reply:
Unresolved this conversation